pkg/util/enablementstatus.go

```diff
@@ -0,0 +1,38 @@
+package util
+
+import "fmt"
+
+// EnablementStatus can be 'enabled', 'disabled' or 'auto'.
+type EnablementStatus int
+
+const (
+	EnablementStatusDisabled EnablementStatus = iota
+	EnablementStatusEnabled  EnablementStatus = iota
+	EnablementStatusAuto     EnablementStatus = iota
+)
+
+func (e *EnablementStatus) String() string {
+	switch *e {
+	case EnablementStatusDisabled:
+		return "disabled"
+	case EnablementStatusEnabled:
+		return "enabled"
+	case EnablementStatusAuto:
+		return "auto"
+	default:
+		panic(fmt.Sprintf("unknown EnablementStatus value %v", *e))
+	}
+}
+
+func (e *EnablementStatus) Set(s string) error {
```

Comment:
This functionality makes me think that we could make `EnablementStatus` a string alias instead of an int alias and then enumerate the strings instead of having the conversion in the `String()` method? I think I would be in favor of making this type a string alias instead let me know your thoughts :thinking:

Reply:
That would work, too. But I have two reasons against:
- the "boring" golang pattern for enums (at least the one I always use) works on ints, and I think boring code is good code - if you can demonstrate that the pattern you're proposing is more "boring", I'm fine to switch to it
- using strings as the backing type would make it possible to abuse this type by casting strings to EnablementStatuses, potentially omitting validation. This is a thread to type safety IMO.

Reply:
Fair enough :+1: